test/api_getProduct_test.dart

```diff
@@ -763,5 +772,86 @@ void main() {
       expect(vegetableFat.vegetarian, IngredientSpecialPropertyStatus.POSITIVE);
       expect(vegetableFat.fromPalmOil, IngredientSpecialPropertyStatus.MAYBE);
     });
+
+    test('matched product', () async {
+      final Map<String, String> attributeImportances = {};
+      int refreshCounter = 0;
+      final ProductPreferencesManager manager = ProductPreferencesManager(
+        ProductPreferencesSelection(
+          (String attributeId, String importanceIndex) async {
+            attributeImportances[attributeId] = importanceIndex;
+          },
+          (String attributeId) =>
+              attributeImportances[attributeId] ??
+              PreferenceImportance.ID_NOT_IMPORTANT,
+          () => refreshCounter++,
+        ),
+      );
+      const String languageCode = 'en';
+      const int httpOk = 200;
+      final String importanceUrl =
+          AvailablePreferenceImportances.getUrl(languageCode);
+      final String attributeGroupUrl =
+          AvailableAttributeGroups.getUrl(languageCode);
+      http.Response response;
+      response = await http.get(Uri.parse(importanceUrl));
+      expect(response.statusCode, httpOk);
+      final String preferenceImportancesString = response.body;
+      response = await http.get(Uri.parse(attributeGroupUrl));
+      expect(response.statusCode, httpOk);
+      final String attributeGroupsString = response.body;
+      manager.availableProductPreferences =
+          AvailableProductPreferences.loadFromJSONStrings(
```

Comment:
Same here, named parameters would be useful to know which string comes first.

Reply:
Good point. I'll do that too.